Library/Homebrew/cmd/update-bash.sh

```diff
@@ -21,12 +21,17 @@ which_git() {
 }
 
 git_init_if_necessary() {
+  set -e
+  trap '{ rm -rf .git; exit 1; }' EXIT
+
   if [[ ! -d ".git" ]]
   then
     git init -q
     git config --bool core.autocrlf false
     git config remote.origin.url https://github.com/Homebrew/homebrew.git
     git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+    git fetch origin
+    git reset --hard origin/master
```

Comment:
```patch
diff --git a/Library/Homebrew/cmd/update-bash.sh b/Library/Homebrew/cmd/update-bash.sh
index 47b811e..4f3adad 100644
--- a/Library/Homebrew/cmd/update-bash.sh
+++ b/Library/Homebrew/cmd/update-bash.sh
@@ -142,6 +142,11 @@ pull() {
   TAP_VAR=$(repo_var "$DIR")
   unset STASHED
 
+  if [[ -z "$(git symbolic-ref refs/remotes/origin/HEAD 2>/dev/null)" ]]
+  then
+    git reset --hard origin/master
+  fi
+
   # The upstream repository's default branch may not be master;
   # check refs/remotes/origin/HEAD to see what the default
   # origin branch name is, and use that. If not set, fall back to "master".
```

This works for me instead of the  `git fetch` and `git reset` additions here.